### PR TITLE
fix: rules engine triggers with duplicates

### PR DESCRIPTION
**Pull request by fpaul-1A**
## Related issues

- :bug: Fixes #(issue)
- :rocket: Feature #(issue)

### CODE REVIEW

**Review by matthieu-crouzet — APPROVED** [review]

**Review by matthieu-crouzet — APPROVED** [review]

File: packages/@o3r/rules-engine/src/services/rules-engine.service.ts [hunk 1]
```diff
@@ -86,7 +86,8 @@ export class RulesEngineService implements OnDestroy {
 
     this.subscription.add(
       this.store.pipe(
-        select(selectAllRulesets)
+        select(selectAllRulesets),
+        distinctUntilChanged((a, b) => JSON.stringify(a) === JSON.stringify(b))
```
Comment (kpanot) [inline packages/@o3r/rules-engine/src/services/rules-engine.service.ts:90]: Why do you need it?
The purpose of having memoize selector is to not have to do this kind of trick :S

Comment (fpaul-1A) [reply packages/@o3r/rules-engine/src/services/rules-engine.service.ts:90]: the store emits twice with the same value, not sure why

Comment (kpanot) [reply packages/@o3r/rules-engine/src/services/rules-engine.service.ts:90]: Would be interesting to find the reason of this double emission actually :S

Comment (fpaul-1A) [reply packages/@o3r/rules-engine/src/services/rules-engine.service.ts:90]: After further investigation, a new event is emitted every time the `updateReducers` is called on ngrx side (for each `StoreModule.forFeature` that is imported for example)
The value emitted seems to be a deepCopy of the same object every time

Comment (fpaul-1A) [reply packages/@o3r/rules-engine/src/services/rules-engine.service.ts]: The multiple emit comes from the default `mergeReducers` function from store-sync creating deep copy of the store
I've updated it to only create deep copy of the store entries that need to be merged
so I removed this `distinctUntilChanged` (assuming people use our fork of store-sync or override the default mergeReducers function accordingly)

**Review by kpanot — APPROVED** [review]

**Review by matthieu-crouzet — APPROVED** [review]

**Pull request merged**